packages/api/src/services/groups.ts

```diff
@@ -172,19 +172,25 @@ export const leaveGroup = async (
       throw new Error('User not in group')
     }
 
+    const membershipId = membership.id
     await t.getRepository(GroupMembership).remove(membership)
 
+    if (group.members.length === 1) {
+      // delete the group if there are no more members
+      await t.getRepository(Group).remove(group)
+
+      return true
+    }
+
     if (membership.isAdmin) {
-      // If the user is the admin, we need to promote another user to admin
+      // If the user is the only admin, we need to promote another user to admin
       const hasAdmin = group.members.some(
-        (m) => m.isAdmin && m.user.id !== user.id
+        (m) => m.isAdmin && m.id !== membershipId
       )
       if (!hasAdmin) {
         const newAdmin = group.members.find((m) => !m.isAdmin)
         if (!newAdmin) {
-          // delete the group if there are no more members
-          await t.getRepository(Group).delete({ id: group.id })
-          return true
+          throw new Error('No user found')
```

Comment:
So it wont delete the group anymore? (That is ok with me for now)

Reply:
It will still delete the group (I moved the code to the other place)

Reply:
@jacksonh